cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/datastore/DatastoreImpl.java

```diff
@@ -1163,18 +1164,22 @@ public Object call(SQLDatabase db) throws Exception {
                         long docNumericId = getNumericIdInQueue(db, item.rev.getId());
                         long seq = 0;
 
+                        InsertRevisionCallable.Result result;
                         if (docNumericId != -1) {
-                            seq = doForceInsertExistingDocumentWithHistory(db, item.rev, docNumericId, item.revisionHistory,
+                             result = doForceInsertExistingDocumentWithHistory(db, item.rev, docNumericId, item.revisionHistory,
                                     item.attachments);
-                            item.rev.initialiseSequence(seq);
                             // TODO fetch the parent doc?
                             documentUpdated = new DocumentUpdated(null, item.rev);
                         } else {
-                            seq = doForceInsertNewDocumentWithHistory(db, item.rev, item.revisionHistory);
-                            item.rev.initialiseSequence(seq);
+                            result  = doForceInsertNewDocumentWithHistory(db, item.rev, item.revisionHistory);
                             documentCreated = new DocumentCreated(item.rev);
                         }
 
+                        item.rev.initialiseSequence(result.sequence);
```

Comment:
I think the call to `initialiseSequence` was back up there so that the `document{Created,Updated}` event would have the correct sequence on it.

Also, we should probably check if it's a duplicate before deciding whether to publish the event.